proposals/networkservicemesh.adoc

```diff
@@ -0,0 +1,95 @@
+== Network Service Mesh Proposal
+
+*Name of project:* Network Service Mesh
+
+*Description*
+
+Network Service Mesh is a novel approach to solving complicated L2/L3 use cases in Kubernetes that are difficult or impossible to address with the existing Kubernetes Network Model. Inspired by Istio, Network Service Mesh maps the concept of a service mesh to L2/L3 payloads.
+
+Network Service Mesh recognizes and respects that the existing Kubernetes Networking provides excellent service for common developer use cases.
+
+Network Service Mesh requires no changes to the CNI plugin being used, or to Kubernetes itself to function.  It runs as an additional layer of infrastructure on top of stock out of the box Kubernetes.
+
```

Comment:
Network Service Mesh provides seamless Layer 2/3 interconnection for a variety of use cases across a single cloud or distributed clouds including infrastructure performance awareness for applications with stringent SLAs [.](https://docs.google.com/document/d/1kibZLSd4BVcnyaVH8hjOJCkELg3BptqIgyAGk7K0NXo/)

Reply:
> Network Service Mesh provides seamless Layer 2/3 interconnection for a variety of use cases across a single cloud or link:https://docs.google.com/document/d/1kibZLSd4BVcnyaVH8hjOJCkELg3BptqIgyAGk7K0NXo/ [distributed clouds] including infrastructure performance awareness for applications with stringent SLAs.

Can we href the usecase doc to distributed clouds.  This will help people to relate NSM with more usecases.